src/librustc_codegen_ironox/type_.rs

```diff
@@ -199,9 +235,14 @@ impl BaseTypeMethods<'tcx> for CodegenCx<'ll, 'tcx> {
 
     fn val_ty(&self, v: Value) -> Type {
         // FIXME: return the real type
-        let mut borrowed_types = self.types.borrow_mut();
-        borrowed_types.push(LLType::None);
-        borrowed_types.len() - 1
+        let mut types = self.types.borrow_mut();
+        match v {
```

Comment:
This might be easier as `if let Value::ConstStruct(_) = v { unimplemented(...) }`.